manifests/install.pp

```diff
@@ -1,15 +1,15 @@
 # Install the foreman proxy
 class foreman_proxy::install {
   if ! $foreman_proxy::custom_repo {
-    foreman::repos { 'foreman_proxy':
+    class { '::foreman::repo':
```

Comment:
Yeah, sorry.  I thought it was reliable, but you're correct, I tried adding both classes to a spec test and it does fail with a duplicate resource - even with https://github.com/theforeman/puppet-foreman/commit/2f2684ee4ad90f0c4a7f2f7dbda2c09562838e7a.

<pre>
  2) foreman_proxy on centos-7-x86_64 with foreman should compile into a catalogue without dependency cycles
     Failure/Error: it { is_expected.to compile.with_all_deps }
       error during compilation: Duplicate declaration: Class[Foreman::Repo] is already declared; cannot redeclare at foreman_proxy/spec/fixtures/modules/foreman_proxy/manifests/install.pp:7 on node cobalt
</pre>

My only suggestion now is to revert the change to share foreman::repo, because if we simply `include` it here, there's no easy way to override `$repo` or `$gpgcheck` if using this foreman_proxy module alone.